### DISCUSSION TITLE
Table flickering?

### DISCUSSION THREAD

**Discussion by denisgurev** [pocketbase/pocketbase#660] [Q&A] [answered]
When you click on the reload button or on some of the sort table headers in the admin, the data table changes the columns size and "flicker/jump" until the new data is loaded.

This is not exactly an issue but it is a little annoying and would be great if it could be prevented by adding a fixed width to the columns or something else.

**Comment by ganigeorgiev** [comment, marked as answer]
I've already "fixed" this while working on the UI changes in relation to the users refactoring in #376.

The flickering happens because we currently reset/clear the table before loading the new data. The "fix" now shows a small loader in the top-left corner (when there is existing data) and clears the table only after the new data has been fetched (there will still be some column size changes due to the dynamic nature of the field values but it is a lot less "jumpy"). If you are curious, it looks something like this:
![screenshot](https://i.imgur.com/XSYmmY9.png)

Other minor improvements to the listings were also implemented like collection fields reordering and overall a better rendering performance, especially noticeable when working with large datasets or collection with a lot of fields/columns.

_The changes will be soon available with the next v0.8.0 release (I still have a lot of integration tests remaining, but I'll try in the next ~2-3 weeks to push at least a pre-release version to unblock some of the closed PRs and other user contributions)._

**Reply by ganigeorgiev** [reply]
@denisgurev Just to let you know that I've cherry-picked some of the UI changes from the upcoming v0.8.0 into the latest [v0.7.7 bugfix release](https://github.com/pocketbase/pocketbase/releases/tag/v0.7.7).
